src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortingType: SortField | null,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortingType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortingType) {
+        case SortField.alphabet:
+          return good1.localeCompare(good2);
+        case SortField.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortingType, setSortingType] = useState<SortField | null>(null);
+  const [reversedState, setReversedState] = useState(false);
+  let visibleGoods = getPreparedGoods(goodsFromServer, sortingType);
+  const resetStates = () => {
+    setSortingType(null);
+    setReversedState(false);
+  };
+
+  if (reversedState) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortingType !== SortField.alphabet,
+          })}
+          onClick={() => setSortingType(SortField.alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortingType !== SortField.length,
+          })}
+          onClick={() => setSortingType(SortField.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !reversedState,
+          })}
+          onClick={() => setReversedState(!reversedState)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortingType || reversedState) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={resetStates}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map(good => (
+          <li key={good} data-cy="Good">
+            {good}
+          </li>
+        ))}
```

Comment:
[CODE KNOWLEDGE] - You have correctly used the `key` prop when rendering the list of goods. This is a good practice and helps React optimize the rendering process.